### PR TITLE
[Serve] Add test case for Serve logs and output

### PR DESCRIPTION
**Pull request by simon-mo**
Need to make sure to merge master to sync with latest output.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/tests/test_standalone.py [hunk 1]
```diff
@@ -508,5 +508,47 @@ def crash():
...
+    def say_hi(self):
+        return "hello"
+
+A.deploy()
+
+resp = requests.get("http://localhost:8000/A")
+resp.raise_for_status()
+
+A.delete()
+serve.shutdown()
```
Comment (architkulkarni) [inline python/ray/serve/tests/test_standalone.py:534]: Does it make sense to add a short sleep after `shutdown`?  Sometimes when I run scripts I notice that messages come after the script exits, maybe the sleep would help capture these.

**Review by architkulkarni — APPROVED** [review]

**Comment by simon-mo** [comment]
@architkulkarni good point, in fact it's already catching it!
```


>       assert len(output) == len(expected_output), "\n".join(output)
--
  | E       AssertionError: 2021-09-23 18:54:52,733	INFO services.py:1252 -- View the Ray dashboard at http://127.0.0.1:8265
  | E         2021-09-23 18:54:52,734	WARNING services.py:1739 -- WARNING: The object store is using /tmp instead of /dev/shm because /dev/shm has only 2684354560 bytes available. This will harm performance! You may be able to free up space by deleting files in /dev/shm. If you are inside a Docker container, you can increase /dev/shm size by passing '--shm-size=4.44gb' to 'docker run' (or add it to the run_options list in a Ray cluster config). Make sure to set this to more than 30% of available RAM.
  | E         (pid=30975) 2021-09-23 18:54:54,850	INFO checkpoint_path.py:15 -- Using RayInternalKVStore for controller checkpoint and recovery.
  | E         (pid=30975) 2021-09-23 18:54:54,871	INFO http_state.py:78 -- Starting HTTP proxy with name 'SERVE_CONTROLLER_ACTOR:UEFiRy:SERVE_PROXY_ACTOR-node:172.18.0.3-0' on node 'node:172.18.0.3-0' listening on '127.0.0.1:8000'
  | E         (pid=30972) INFO:     Started server process [30972]
  | E         2021-09-23 18:54:55,240	INFO api.py:455 -- Started Serve instance in namespace 'serve'.
  | E         2021-09-23 18:54:55,259	INFO api.py:243 -- Updating deployment 'A'. component=serve deployment=A
  | E         (pid=30975) 2021-09-23 18:54:55,295	INFO backend_state.py:896 -- Adding 1 replicas to deployment 'A'. component=serve deployment=A
  | E         2021-09-23 18:54:55,822	INFO api.py:251 -- Deployment 'A' is ready at `http://127.0.0.1:8000/A`. component=serve deployment=A
  | E         (pid=30975) 2021-09-23 18:54:55,923	INFO backend_state.py:914 -- Removing 1 replicas from deployment 'A'. component=serve deployment=A
  | E         (pid=30974) 2021-09-23 18:54:57,928	ERROR backend_worker.py:361 -- Exception during graceful shutdown of replica. component=serve deployment=A replica=A#auNujE
  | E         (pid=30974) Traceback (most recent call last):
  | E         (pid=30974)   File "/ray/python/ray/serve/backend_worker.py", line 359, in prepare_for_shutdown
  | E         (pid=30974)     self.callable.__del__()
  | E         (pid=30974)   File "/ray/python/ray/serve/api.py", line 605, in __del__
  | E         (pid=30974)     self._serve_asgi_lifespan.shutdown())
  | E         (pid=30974)   File "/opt/miniconda/lib/python3.6/asyncio/base_events.py", line 475, in run_until_complete
  | E         (pid=30974)     self.run_forever()
  | E         (pid=30974)   File "/opt/miniconda/lib/python3.6/asyncio/base_events.py", line 429, in run_forever
  | E         (pid=30974)     raise RuntimeError('This event loop is already running')
  | E         (pid=30974) RuntimeError: This event loop is already running
  | E         (pid=30974) 2021-09-23 18:54:57,929	INFO on.py:64 -- Waiting for application shutdown. component=serve deployment=A replica=A#auNujE
  | E         (pid=30974) 2021-09-23 18:54:57,929	INFO on.py:75 -- Application shutdown complete. component=serve deployment=A replica=A#auNujE
  | E       assert 23 == 8
  | E         +23
  | E         -8
```

**Comment by architkulkarni** [comment]
Another approach, which might be less rigid and more future-proof, would be to just search the output for `"warning"` or `"error"`.  It seems reasonable to require that the output of a basic Serve script should never contain `"warning"` or `"error"`.  But I think the current approach is fine as well

**Review by edoakes — APPROVED** [review]

**Comment by bveeramani** [comment]
## :bangbang: ACTION REQUIRED :bangbang:

We've switched our code formatter from YAPF to Black (see #21311).

To prevent issues with merging your code, here's what you'll need to do:

1. Install Black
```
pip install -I black==21.12b0
```

2. Format changed files with Black
```
curl -o format-changed.sh https://gist.githubusercontent.com/bveeramani/42ef0e9e387b755a8a735b084af976f2/raw/7631276790765d555c423b8db2b679fd957b984a/format-changed.sh
chmod +x ./format-changed.sh
./format-changed.sh
rm format-changed.sh
```

3. Commit your changes.
```
git add --all
git commit -m "Format Python code with Black"
```

4. Merge master into your branch.
```
git pull upstream master
```

5. Resolve merge conflicts (if necessary).

After running these steps, you'll have the updated `format.sh`.

**Comment by simon-mo** [comment]
Yeah... I have fighting against the CI for this one.

**Pull request closed without merging**